pydatastructs/trees/m_ary_trees.py

```diff
@@ -0,0 +1,168 @@
+from pydatastructs.utils import MAryTreeNode
+from pydatastructs.linear_data_structures.arrays import ArrayForTrees
+
+__all__ = [
+    'MAryTree'
+]
+
+class MAryTree(object):
+    """
+    Abstract m-ary tree.
+
+    Parameters
+    ==========
+
+    max_children
+        Optional, specifies the maximum number of children
+        a node can have. Defaults to 2 in case nothing is
+        specified.
+    root_data
+        Optional, the root node of the binary tree.
+        If not of type MAryTreeNode, it will consider
+        root as data and a new root node will
+        be created.
+    key
+        Required if tree is to be instantiated with
+        root otherwise not needed.
+    comp: lambda
+        Optional, A lambda function which will be used
+        for comparison of keys. Should return a
+        bool value. By default it implements less
+        than operator.
+    is_order_statistic: bool
+        Set it to True, if you want to use the
+        order statistic features of the tree.
```

Comment:
```suggestion
    key
        Required if tree is to be instantiated with
        root otherwise not needed.
    root_data
        Optional, the root node of the binary tree.
        If not of type MAryTreeNode, it will consider
        root as data and a new root node will
        be created.
    comp: lambda
        Optional, A lambda function which will be used
        for comparison of keys. Should return a
        bool value. By default it implements less
        than operator.
    is_order_statistic: bool
        Set it to True, if you want to use the
        order statistic features of the tree.
    max_children
        Optional, specifies the maximum number of children
        a node can have. Defaults to 2 in case nothing is
        specified.
```

Reply:
Changed.